extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -144,6 +169,36 @@ export class AmpAd3PImpl extends AMP.BaseElement {
     this.uiHandler = new AmpAdUIHandler(this);
 
     setupA2AListener(this.win);
+
+    this.computeFullWidthState_();
+
+    if (this.fullWidthState_ == FullWidthState.REQUESTED) {
+      return this.attemptFullWidthSizeChange_();
+    }
+  }
+
+  /**
+   * Sets the fullWidthState_ variable.
+   * @private
+   */
+  computeFullWidthState_() {
+    const hasFullWidth = this.element.hasAttribute('data-full-width');
+    if (!hasFullWidth) {
+      this.fullWidthState_ = FullWidthState.NOT_ENABLED;
+      return;
+    }
+    user().assert(this.element.getAttribute('width') == '100vw',
```

Comment:
shall we just override width to 100vw if we see `data-full-width`?

Reply:
Are we allowed to always do that successfully at this point?
If not let's not do it - from a product perspective I think it's better for us to consistently fail than to work in some cases.